prime-dotnet-webapi/Services/EnrolleeService.cs

```diff
@@ -954,5 +954,51 @@ public async Task<IEnumerable<string>> GetEnrolleeEmails(BulkEmailType bulkEmail
                 .DecompileAsync()
                 .ToListAsync();
         }
+
+        public async Task<bool> IsPotentialPaperEnrolleeReturnee(string dateOfBirth)
+        {
+            return await _context.Enrollees
+                .AsNoTracking()
+                .AnyAsync(e => e.GPID.StartsWith("NOBCSC")
+                    && e.DateOfBirth.Date == DateTime.Parse(dateOfBirth).Date
+                    && e.LinkedErolmentId == 0
+                );
+        }
+
+        public async Task<IEnumerable<Enrollee>> GetPotentialPaperEnrolleeReturnees(DateTime dateOfBirth)
+        {
+            return await _context.Enrollees
+                .AsNoTracking()
+                .Where(e => e.GPID.StartsWith("NOBCSC")
+                    && e.DateOfBirth.Date == dateOfBirth.Date
+                    && e.LinkedErolmentId == 0
+                )
+                .ToListAsync();
+        }
+
+        public async Task<bool> LinkEnrolmentToPaperEnrolment(int enrolmentId, int PaperEnrolmentId)
+        {
+            var enrollee = await _context.Enrollees
+                .Where(e => e.Id == enrolmentId)
+                .SingleOrDefaultAsync();
+
+            var paperEnrollee = await _context.Enrollees
+                .Where(pe => pe.GPID.StartsWith("NOBCSC") && pe.Id == PaperEnrolmentId)
+                .SingleOrDefaultAsync();
+
+            int DbLinkedEnrolmentId = enrollee.LinkedErolmentId;
+            int DbLinkedPaperEnrolmentId = paperEnrollee.LinkedErolmentId;
+
+            if (DbLinkedEnrolmentId != 0 || DbLinkedPaperEnrolmentId != 0)
+            {
+                return false;
+            }
+            else
+            {
+                _context.Update(DbLinkedEnrolmentId);
+                _context.Update(DbLinkedPaperEnrolmentId);
+            }
+            return true;
+        }
```

Comment:
I feel these methods may be more suitable on the `EnrolleePaperSubmissionService` instead.

Reply:
Moved to EnrolleePaperSubmissionService